### PR TITLE
Update DiskUsage to support new format

### LINKED ISSUE #687: `DiskUsage` plugin should be able to consume new data format
...wrt. disk usage report.

### PR DESCRIPTION
**Pull request by asnyv**
Note that the new format is still in testing, and not fully deployed. Therefore supports the old format as well.
Should we leave it out of the changelog, considering users (hopefully) will not notice the change?

---

### Contributor checklist

- [X] :tada: This PR closes #687 .
- [X] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR: updates to testdata in: https://github.com/equinor/webviz-subsurface-testdata/pull/90
- [X] :book: I have considered adding a new entry in `CHANGELOG.md`, and added it if should be communicated there.

### CODE REVIEW

File: webviz_subsurface/plugins/_disk_usage.py [hunk 1]
```diff
@@ -127,56 +126,87 @@ def add_webvizstore(self) -> List[Tuple[Callable, list]]:
...
+        f"No disk usage file found for the last week in {scratch_dir}."
+    )
+
+
+def _estimate_free_space(df: pd.DataFrame, scratch_dir: Path, date: str) -> float:
+    txt_file = Path(scratch_dir / ".disk_usage" / f"disk_usage_user_test_{date}.txt")
+    total_usage = None
+    if txt_file.exists():
+        with open(txt_file, "r") as filehandle:
+            lines = filehandle.readlines()
```
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_disk_usage.py:204]:
```suggestion
        lines = txt_file.read_text().splitlines()
```

**Review by anders-kiaer — COMMENTED** [review]
LGTM! 👍 

> Should we leave it out of the changelog, considering users (hopefully) will not notice the change?

Maybe we should add one sentence stating the new format is supported, and what that improves in the visualization/plugin compared with the old format?

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ from the water zone if the volumes from the full grid geometry are included, and
 
 ### Added
 - [#662](https://github.com/equinor/webviz-subsurface/pull/662) - Added support in `WellCompletion` for connection history from summary data.
+- [#708](https://github.com/equinor/webviz-subsurface/pull/708) - Added support for new report format for `DiskUsage`, which improves the estimate of free disk space.
```
Comment (anders-kiaer) [inline CHANGELOG.md:29]: Need to move this now since we waited with merge.

File: webviz_subsurface/plugins/_disk_usage.py [hunk 3]
```diff
@@ -127,56 +126,86 @@ def add_webvizstore(self) -> List[Tuple[Callable, list]]:
...
             ignore_index=True,
         )
 
-    return df.sort_values(by="usageKB", axis=0, ascending=False)
+    df["date"] = date
+    return df.sort_values(by="usageGiB", axis=0, ascending=False)
+
+
+def _get_disk_usage_for_date(scratch_dir: Path, date: str) -> Optional[pd.DataFrame]:
+    csv_file = Path(scratch_dir / ".disk_usage" / f"disk_usage_user_{date}.csv")
```
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_disk_usage.py:181]:
```suggestion
    csv_file = scratch_dir / ".disk_usage" / f"disk_usage_user_{date}.csv"
```

**Review by anders-kiaer — CHANGES_REQUESTED** [review]
:+1: Two small suggestions, otherwise looks ready for merge to me. :slightly_smiling_face:

File: CHANGELOG.md (hunk 2 above)
Comment (asnyv) [reply CHANGELOG.md:29]: Hmm, it is still under UNRELEASED, but seems like we have 2 `Added` sections (Added - Changed - Fixed - Added). 
Do we have a standard for the order of `Added`, `Changed` and `Fixed`? The natural is maybe that order (alphabetical), but e.g. for the `0.2.3` and `0.2.1` releases it was `Changed`, `Fixed`, `Added`.

Comment (asnyv) [reply CHANGELOG.md:29]: sorry, latest release is `0.2.4 `now 😅 and hadn't rebased......

**Review by anders-kiaer — APPROVED** [review]

**Pull request merged**